plugins/package-managers/spm/src/main/kotlin/SpmModel.kt

```diff
@@ -136,35 +131,25 @@ data class AppDependency(
 
     override val id: Identifier
         get() {
-            val (author, _) = parseAuthorAndProjectFromRepo(repositoryUrl)
             return Identifier(
                 type = PACKAGE_TYPE,
-                namespace = author.orEmpty(),
-                name = packageName,
+                namespace = "",
+                name = getCanonicalName(repositoryUrl),
                 version = state?.toString().orEmpty()
             )
         }
 }
 
-private val logger = loggerOf(MethodHandles.lookup().lookupClass())
-
-internal fun parseAuthorAndProjectFromRepo(repositoryURL: String): Pair<String?, String?> {
-    val normalizedURL = normalizeVcsUrl(repositoryURL)
-    val vcsHost = VcsHost.fromUrl(URI(normalizedURL))
-    val project = vcsHost?.getProject(normalizedURL)
-    val author = vcsHost?.getUserOrOrganization(normalizedURL)
-
-    if (author.isNullOrBlank()) {
-        logger.warn {
-            "Unable to parse the author from VCS URL $repositoryURL, results might be incomplete."
-        }
-    }
-
-    if (project.isNullOrBlank()) {
-        logger.warn {
-            "Unable to parse the project from VCS URL $repositoryURL, results might be incomplete."
-        }
-    }
-
-    return author to project
+/**
+ * Return the canonical name for a package based on the given [repositoryUrl].
+ * The algorithm assumes that the repository URL does not point to the local file
+ * system, as support for local dependencies is not implemented yet in ORT. Otherwise,
+ * the algorithm tries to effectively mimic the algorithm described in
+ * https://github.com/apple/swift-package-manager/blob/24bfdd180afdf78160e7a2f6f6deb2c8249d40d3/Sources/PackageModel/PackageIdentity.swift#L345-L415.
+ */
+internal fun getCanonicalName(repositoryUrl: String): String {
+    val normalizedUrl = normalizeVcsUrl(repositoryUrl)
+    return normalizedUrl.toUri {
+        "${it.host}${it.path.removeSuffix(".git")}"
```

Comment:
Nit: I believe `it.host + it.path.removeSuffix(".git")` would be a bit more readable as it avoids nesting of double quotes.

Reply:
I'll fix this with the following PR, as after going through a lot of flaky CI re-runs yesterday scared me of going through it again.

Reply:
here you go: https://github.com/oss-review-toolkit/ort/pull/8137